release/pypi/prep_binary_for_pypi.py

```diff
@@ -0,0 +1,169 @@
+
+        # Use InWheelCtx to work with the wheel contents directly
+        with InWheelCtx(whl_file) as ctx:
+            ctx.out_wheel = tmp_whl
+            # Find the .dist-info directory
+            dist_info_dirs = [
+                d
+                for d in os.listdir(ctx.path)
+                if d.endswith(".dist-info") and os.path.isdir(os.path.join(ctx.path, d))
+            ]
+
+            if not dist_info_dirs:
+                print(f"Error: No .dist-info directory found in {whl_file}")
+                return
+
+            dist_info_dir = dist_info_dirs[0]
+
+            # Read the metadata to get version info
+            metadata_file = os.path.join(ctx.path, dist_info_dir, "METADATA")
+            if not os.path.exists(metadata_file):
+                print(f"Error: METADATA file not found in {whl_file}")
+                return
+
+            # Extract version with suffix
+            version_with_suffix = None
+            with open(metadata_file, "r", encoding="utf-8") as f:
+                for line in f:
+                    if line.startswith("Version:"):
+                        version_with_suffix = line.split(":", 1)[1].strip()
+                        break
+
+            if not version_with_suffix:
+                print(f"Error: Could not find version in {metadata_file}")
+                return
+
+            # Check if there's a suffix to remove
+            if "+" not in version_with_suffix:
+                print(f"No suffix found in version {version_with_suffix}, skipping")
+                return
+
+            # Remove suffix from version
+            version_no_suffix = version_with_suffix.split("+")[0]
+            print(f"Removing suffix: {version_with_suffix} -> {version_no_suffix}")
+
+            # Update version in all files in dist-info
+            for root, _dirs, files in os.walk(os.path.join(ctx.path, dist_info_dir)):
+                for file in files:
+                    file_path = os.path.join(root, file)
+                    try:
+                        with open(
+                            file_path, "r", encoding="utf-8", errors="ignore"
+                        ) as f:
+                            content = f.read()
+
+                        # Replace version with suffix to version without suffix
+                        updated_content = content.replace(
+                            version_with_suffix, version_no_suffix
+                        )
+
+                        if content != updated_content:
+                            with open(file_path, "w", encoding="utf-8") as f:
+                                f.write(updated_content)
+                    except UnicodeDecodeError:
+                        # Skip binary files
+                        pass
+
+            # Rename the dist-info directory
+            new_dist_info_dir = dist_info_dir.replace(
+                version_with_suffix, version_no_suffix
+            )
```

Comment:
nit: Might be good to include an example of what version_with_suffix --> version_no_suffix looks like.